### PR TITLE
nodePackages/ssb-room: init at 1.1.2

### PR DESCRIPTION
**Pull request by asymmetric**
##### Motivation for this change

Add [ssb-room](https://github.com/staltz/ssb-room) package.

###### Things done

Ran the `./generate.sh` script after adding the package to `node-packages-v10.json`.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

/cc @NinjaTrappeur

### CODE REVIEW

File: pkgs/development/node-packages/composition-v13.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 {pkgs ? import <nixpkgs> {
     inherit system;
-  }, system ? builtins.currentSystem, nodejs ? pkgs."nodejs-13_x"}:
+  }, system ? builtins.currentSystem, nodejs ? pkgs."nodejs-8_x"}:
```
Comment (asymmetric) [inline pkgs/development/node-packages/composition-v13.nix]: Not sure why this got changed.

File: pkgs/development/node-packages/node-env.nix [hunk 2]
```diff
@@ -363,7 +363,7 @@ let
 
         npm ${forceOfflineFlag} --nodedir=${nodeSources} ${npmFlags} ${stdenv.lib.optionalString production "--production"} rebuild
 
-        if [ "''${dontNpmInstall-}" != "1" ]
+        if [ "$dontNpmInstall" != "1" ]
```
Comment (asymmetric) [inline pkgs/development/node-packages/node-env.nix]: Nor this.

File: pkgs/development/node-packages/composition-v13.nix (hunk 1 above)
Comment (NinjaTrappeur) [reply pkgs/development/node-packages/composition-v13.nix]: I was about to ask :)

File: pkgs/development/node-packages/node-env.nix (hunk 2 above)
Comment (NinjaTrappeur) [reply pkgs/development/node-packages/node-env.nix]: This comes from a tree-wide change: https://github.com/NixOS/nixpkgs/commit/2811b032d639684d3983282e9d10d89f37d161d5

It is to be expected.

@Ericson2314, this file has been auto-generated, should we upstream that to https://github.com/svanderburg/node2nix/blob/master/nix/node-env.nix#L366 ?

File: pkgs/development/node-packages/composition-v13.nix (hunk 1 above)
Comment (NinjaTrappeur) [reply pkgs/development/node-packages/composition-v13.nix]: `node2nix` is called with the `--node-js-13` argument, which according to https://github.com/svanderburg/node2nix/blob/master/bin/node2nix.js#L134 is not implemented yet...

@marsam did you created this file manually? Is anything special preventing us to upstream that to `node2nix`?

Comment (NinjaTrappeur) [reply pkgs/development/node-packages/composition-v13.nix]: Created a proper issue to track that: https://github.com/NixOS/nixpkgs/issues/78044

Comment (NinjaTrappeur) [reply pkgs/development/node-packages/composition-v13.nix]: Created a proper issue to track that: https://github.com/NixOS/nixpkgs/issues/78044

Comment (marsam) [reply pkgs/development/node-packages/composition-v13.nix]: >@marsam did you created this file manually?

yes, I remember changing this line with `13_x`. Sorry about that!

Comment (NinjaTrappeur) [reply pkgs/development/node-packages/composition-v13.nix]: > Sorry about that!

No problem :) Let's fix this :+1:

**Comment by prusnak** [comment]
Please rework your PR. It now has a merge conflict after PR https://github.com/NixOS/nixpkgs/pull/89184 has been merged

**Comment by asymmetric** [comment]
Closing, as I ended up not using this.

**Comment by NinjaTrappeur** [comment]
If somebody want to take over this at some point: the ground work in node2nix has been done. More infos on https://github.com/NixOS/nixpkgs/issues/78044

**Pull request closed without merging**